lib/views/course_view/components/course_card.dart

```diff
@@ -1,144 +1,397 @@
+                width: 100,
+                child: _buildCourseImage(),
+              ),
+              const SizedBox(width: 15),
+              Expanded(
+                child: Column(
+                  crossAxisAlignment: CrossAxisAlignment.start,
                   children: [
+                    const SizedBox(height: 10),
+                    _buildCourseTitle(themeData.textTheme),
                     _buildCourseSubtitle(themeData.textTheme),
-                    _buildCourseIsLive(),
+                    const SizedBox(height: 15),
+                    _buildLocation(),
+                    //_buildCourseIsLive(),
                   ],
                 ),
-                Expanded(child: _buildCourseImage()),
-                _buildCourseTitle(themeData.textTheme),
-              ],
+              ),
+            ],
+          ),
+        ],
+      ),
+    );
+  }
+
+  Widget _buildCourseCard(
+    ThemeData themeData,
+    double cardWidth,
+    BuildContext context,
+  ) {
+    return IntrinsicHeight(
+      child: Row(
+        children: [
+          _buildCourseColor(),
+          Expanded(
+            child: Container(
+              color: themeData.cardColor,
+              padding: const EdgeInsets.all(8.0),
+              child: Column(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: [
+                  Row(
+                    mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                    children: [
+                      _buildCourseTumID(),
+                      _buildCourseIsLive(),
+                    ],
+                  ),
+                  Padding(
+                    padding: const EdgeInsets.symmetric(vertical: 3.0),
+                    child: _buildCourseTitle(themeData.textTheme),
+                  ),
+                  _buildLastLecture(context),
+                ],
+              ),
             ),
           ),
-        ),
+        ],
       ),
     );
   }
 
   Widget _buildCourseImage() {
+    if (path == null) return const SizedBox();
     return Stack(
       children: [
         AspectRatio(
-          aspectRatio: 10 / 7,
+          aspectRatio: 16 / 12, //TODO how are the thumbnails given?
           child: ClipRRect(
             borderRadius: BorderRadius.circular(8.0),
             child: Image.asset(
-              path,
+              path!,
               fit: BoxFit.cover,
             ),
           ),
         ),
+        /*Positioned(
+          bottom: 3, // Adjust this value based on your layout
+          right: 3, // Adjust this value based on your layout
+          child: _buildCourseViewerCount(),
+        ),*/
       ],
     );
   }
 
+  Widget _buildLocation() {
+    if (roomName == null) return const SizedBox();
+
+    final Uri url = roomName != null
+        ? Uri.parse('https://nav.tum.de/room/$roomNumber')
+        : Uri.parse(
+            'https://nav.tum.de/room/',
+          ); //will give an error on NavigaTUM
```

Comment:
What does this mean?

Reply:
The idea was that when the roomnumber is not is not set for a course, we still redirect to NavigaTum with the invalid room number and make them return an "invalid" 

But I will change it so that the section is simply not being displayed in that case